src/js/components/List/List.js

```diff
@@ -245,7 +247,19 @@ const List = React.forwardRef(
             as={as || 'ul'}
             itemFocus={itemFocus}
             tabIndex={onClickItem || onOrder ? 0 : undefined}
-            onBlur={onOrder ? () => setActive(undefined) : undefined}
+            onBlur={
+              onOrder
+                ? () => {
+                    setActive(undefined);
+                  }
+                : () => {
+                    if (active) setLastActive(active);
+                    setActive(undefined);
+                  }
+            }
+            onFocus={() => {
+              if (lastActive) setActive(lastActive);
```

Comment:
If `lastActive` is zero, this won't be true. I think we can always `setActive(lastActive)` irrespective of what `lastActive` is.

Reply:
After logging the relationship between `active and lastActive` , I updated this code chunk to handle what happens when `active` is read as `falsy`.

Now when you click / press enter on the first entry in the `onClickItem` story, remove the table's focus, and then regain the table's focus, the first entry still remains active as intended. 

This also solves the bug where:
- clicking / pressing enter on an item, clicking off the table / removing table focus, regaining table focus, and then holding down a click on another item would show the focus style on the `lastActive` instead of the item the user is currently clicking down on.